pkg/sources/adapter/cloudeventssource/cloudevents_test.go

```diff
@@ -0,0 +1,223 @@
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+package cloudeventssource
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	zapt "go.uber.org/zap/zaptest"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cloudeventst "github.com/cloudevents/sdk-go/v2/client/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/triggermesh/triggermesh/pkg/adapter/fs"
+)
+
+const (
+	// path to secret fixtures.
+	secret1Path = "../../../../test/fixtures/secrets/secret1"
+	secret2Path = "../../../../test/fixtures/secrets/secret2"
+
+	// values for secrets in the fixtures above.
+	tSecret1 = "secret1"
+	tSecret2 = "secret2"
+
+	tUser  = "user"
+	tToken = "token"
+)
+
+var (
+	basicAuths KeyMountedValues = KeyMountedValues{
+		{
+			Key:              tUser,
+			MountedValueFile: secret1Path,
+		},
+	}
+	tokens KeyMountedValues = KeyMountedValues{
+		{
+			Key:              tToken,
+			MountedValueFile: secret2Path,
+		},
+	}
+)
+
+func TestCloudEventsSource(t *testing.T) {
+	logger := zapt.NewLogger(t).Sugar()
+
+	successCE := cloudevents.NewEvent(cloudevents.VersionV1)
+	successCE.SetType("type")
+	successCE.SetSource("source")
+	successCE.SetID("id")
+	successCE.SetTime(time.Now())
+
+	failCE := cloudevents.NewEvent(cloudevents.VersionV1)
+
+	tc := map[string]struct {
+		cloudEvent  cloudevents.Event
+		expectError bool
+	}{
+		"success CE": {
+			cloudEvent:  successCE,
+			expectError: false,
+		},
+		"fail CE": {
+			cloudEvent:  failCE,
+			expectError: true,
+		},
+	}
+
+	for name, c := range tc {
+		t.Run(name, func(t *testing.T) {
+			cfw, err := fs.NewCachedFileWatcher(logger)
+			require.NoError(t, err, "Could not create CachedFileWatcher")
+
+			ceClient, chOut := cloudeventst.NewMockSenderClient(t, 1)
+
+			handler := &cloudEventsHandler{
+				cfw:      cfw,
+				ceClient: ceClient,
+				logger:   logger,
+			}
+
+			res := handler.handle(context.TODO(), c.cloudEvent)
```

Comment:
nit: There is no advantage using `context.TODO()` over `context.Background()` and I remember semgrep [yields a warning](https://github.com/triggermesh/triggermesh/issues/174) when it sees it.

> "Consider to use well-defined context"

Also, it's a bit annoying to exclude this pattern when grepping for `TODO` (but maybe I'm the only one who annotates my code with a bunch of TODO while working on draft code 😅)

Reply:
https://github.com/triggermesh/triggermesh/pull/730/commits/64ac555e2f9c7c4a35d31f2515e79c6bff6da83c